monitor/src/config.rs

```diff
@@ -24,22 +24,30 @@ impl Config {
         let mut contents = String::new();
         file.read_to_string(&mut contents).map_err(|e| format!("Failed to read config file: {}", e))?;
 
-        let config: Config = serde_yaml::from_str(&contents).map_err(|e| format!("Failed to parse YAML: {}", e))?;
+        let mut config: Config = serde_yaml::from_str(&contents).map_err(|e| format!("Failed to parse YAML: {}", e))?;
+        config.file_path = Some(path.to_string());
 
         // Verify that the actions files exist for each configuration
         for single_config in &config.configurations {
-            if !Path::new(&single_config.actions_path).exists() {
-                return Err(format!("File {} not found", single_config.actions_path));
-            }
+            Self::exists_actions_file(&single_config.actions_path, &single_config.repo_name);
         }
-
         Ok(config)
     }
 
-    pub(crate) fn exists_actions_file(actions_path: &String, repo_name: &String) -> Result<(), Box<dyn Error>> {
+    pub(crate) fn exists_actions_file(actions_path: &String, repo_name: &String) {
         if !Path::new(&actions_path).exists() {
-            return Err(format!("The actions file '{}' for repo '{}' does not exist.", actions_path, repo_name).into());
+            panic!("The actions file '{}' for repo '{}' does not exist.", &actions_path, repo_name);
```

Comment:
Manage this error like before: 

```suggestion
        return Err(format!("The actions file '{}' for repo '{}' does not exist.", actions_path, repo_name).into());
```